core/src/utils/prove.rs

```diff
@@ -89,9 +86,123 @@ pub fn run_test(program: Program) -> Result<(), crate::stark::ProgramVerificatio
     Ok(())
 }
 
-pub fn prove_elf(elf: &[u8]) -> crate::stark::Proof<BabyBearBlake3> {
-    let program = Program::from(elf);
-    prove(program)
+fn trace_checkpoint(program: Program, file: &File) -> ExecutionRecord {
+    let mut reader = std::io::BufReader::new(file);
+    let state = bincode::deserialize_from(&mut reader).expect("failed to deserialize state");
+    let (events, _) = tracing::debug_span!("runtime.trace").in_scope(|| {
+        let mut runtime = Runtime::new(program.clone());
+        runtime.state = state;
+        runtime.record = ExecutionRecord::default();
+        runtime.record.program = Arc::new(program.clone());
+        let index: u32 = (runtime.state.global_clk / (runtime.shard_size / 4) as u64)
+            .try_into()
+            .unwrap();
+        runtime.record.index = index;
+        runtime.execute_record()
+    });
+    events
+}
+
+fn reset_seek(file: &mut File) {
+    file.seek(std::io::SeekFrom::Start(0))
+        .expect("failed to seek to start of tempfile");
+}
+
+pub fn run_and_prove<SC: StarkGenericConfig + StarkUtils + Send + Sync + Serialize>(
```

Comment:
Is the behavior actually the same if the env var is set to 0? Won't it execute the runtime twice?

Reply:
yes it will, first time without events though. maybe we should add a special case?

Reply:
Yeah I think it's probably worth doing that if you can do it in a clean way